Tools/PlatformWPE.cmake

```diff
@@ -1,24 +1,37 @@
-if (DEVELOPER_MODE OR ENABLE_MINIBROWSER)
+if (ENABLE_API_TESTS OR ENABLE_LAYOUT_TESTS OR ENABLE_MINIBROWSER)
     add_subdirectory(wpe/backends)
 endif ()
 
-if (DEVELOPER_MODE)
+if (ENABLE_LAYOUT_TESTS)
     add_subdirectory(ImageDiff)
     add_subdirectory(TestRunnerShared)
     add_subdirectory(WebKitTestRunner)
+endif ()
+
+if (DEVELOPER_MODE)
     add_subdirectory(flatpak)
+endif ()
 
-    if (ENABLE_API_TESTS)
-        add_subdirectory(TestWebKitAPI/glib)
-    endif ()
+if (ENABLE_API_TESTS)
+    add_subdirectory(TestWebKitAPI/glib)
 endif ()
 
 if (ENABLE_MINIBROWSER)
     add_subdirectory(MiniBrowser/wpe)
 endif ()
 
-if (DEVELOPER_MODE AND ENABLE_COG)
+if (ENABLE_COG)
     include(ExternalProject)
+    if ("${WPE_COG_PLATFORMS}" STREQUAL "")
+        set(WPE_COG_PLATFORMS "drm,headless,gtk4,x11,wayland")
+    elseif ("${WPE_COG_PLATFORMS}" STREQUAL "none")
+        set(WPE_COG_PLATFORMS "")
+    endif ()
+    if (DEFINED ENV{PKG_CONFIG_PATH})
+        set(WPE_COG_PKG_CONFIG_PATH ${CMAKE_BINARY_DIR}:$ENV{PKG_CONFIG_PATH})
+    else ()
+        set(WPE_COG_PKG_CONFIG_PATH ${CMAKE_BINARY_DIR})
+    endif ()
```

Comment:
Hm, `${CMAKE_BINARY_DIR}` is notably the _toplevel_ build directory. Surely cog.pc will not be installed there. Is that telling pkg-config to search the _entire_ build directory for pkg-config files? Maybe this could be more targeted?

Reply:
It is actually searching for `wpe-webkit-1.0.pc` there. The idea is that we don't install WPE after building it, but instead tell the cog build system to look for the wpe.pc file in the build directory